tests/test_redis_client.py

```diff
@@ -22,6 +27,7 @@ def build_redis_client(app, mocker):
                         return_value={b'template-1111': b'8', b'template-2222': b'8'})
     mocker.patch.object(redis_client.redis_store, 'hmset')
     mocker.patch.object(redis_client.redis_store, 'expire')
+    mocker.patch.object(redis_client.redis_store, 'pipeline', return_value=my_redis_mock)
```

Comment:
one reason we don't want this to be a module level variable, i've just realised, is that `my_redis_mock` won't be reset between calls.  so any checks to say "did this function get called" will be looking at every test that has ran so far, and if a test that doesn't change `my_redis_mock.execute.return_value` runs after one that *does*, it'll use the results from the one that does